### PR TITLE
Break down ExperimentDetails container into (mostly) simple components

### PR DESCRIPTION
**Pull request by lmorchard**
Opening a WIP PR for this, now that I'm a bit into it. It's getting to be a large set of changes, but mostly just shuffling things around. Trying to break things up into multiple commits to capture smaller discrete changes.

Overview of things I've done so far:

* Move components from `components/` folder into `containers/ExperimentPage/` where they are only ever used by the experiment page. Import paths adjusted as necessary.
* Merged ExperimentPage tests into `containers/ExperimentPage/tests.js`. Import paths adjusted as necessary.
* Lint fixes to test code, because it seems that eslint wasn't checking those and a pile of little problems snuck in.
* Trying to apply more flow type annotations, which seems to work better with separate `types.js` modules. Importing both a component class and a type from the same module seems to be a linting error.

Things left to do: (which will be saved for follow up PRs)
- Fill out more stories around sub-components
- Further break up the mass of markup in ExperimentPage down into sub-components

### CODE REVIEW

**Comment by lmorchard** [comment]
Also, pushing this into a branch on the mozilla repo, so we get some Storybook builds along the way.

**Comment by lmorchard** [comment]
I'm also kind of on the fence about this bit from [the reorg proposal](https://gist.github.com/chuckharmston/01e5b7a16ba9b771863cbb8b2de64138), now that I'm in the thick of things:

> Dependent components will be defined in the same file as the parent component. 

I can merge them together before we merge this PR. But, so far, it's feeling a little more manageable to leave dependent components for ExperimentPage in separate files - because there are just so many of them and they do so much. I might also re-split the tests into separate files, for basically the same reason.

**Comment by lmorchard** [comment]
Alright: After some IRC chat, I'm going to take the WIP off this PR and stop adding to it. There's enough here to have a good start on pulling further things out in smaller chunks.

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2891/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/09050ce6bf9f550a2fa222dbe09083698365c7bf/

**Comment by lmorchard** [comment]
This PR is bit rotting fast as it ages. Gave it a big rebase to resolve conflicts

**Comment by lmorchard** [comment]
Also noticed that the new `GraduationNotice` component is only used on `ExperimentPage`. Tempted to move it into the bundle directory too, but also want to resist changing this PR any further

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2891/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/dd6342d046ac07cefe720f5f76fe566ed29cce77/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2891/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/53716fc5dea653e743ea738423ed1065d79a0892/

File: frontend/src/app/components/MainInstallButton.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// @flow
```
Comment (meandavejustice) [inline frontend/src/app/components/MainInstallButton.js:1]: Shouldn't this file live in a directory `frontend/src/app/components/MainInstallButton/index.js`?

File: frontend/src/app/components/types.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {
```
Comment (meandavejustice) [inline frontend/src/app/components/types.js:1]: same question as above but Shouldn't this file live in a directory `frontend/src/app/components/MainInstallButton/types.js`

**Review by meandavejustice — COMMENTED** [review]
I like the way you've divided these up. I only had two comments about the structure of files, I wouldn't hold up a merge over those, but interested in your reasoning.

File: frontend/src/app/components/MainInstallButton.js (hunk 1 above)
Comment (lmorchard) [reply frontend/src/app/components/MainInstallButton.js:1]: Eventually, yeah, but I didn't want to spread out into too many changes while focusing on ExperimentPage

File: frontend/src/app/components/types.js (hunk 2 above)
Comment (lmorchard) [reply frontend/src/app/components/types.js:1]: Maybe, but for now it has types for multiple components and not just MainInstallButton

**Comment by fzzzy** [comment]
Awesome that this is merged!

**Pull request merged**